lib/fastlane/plugin/wpmreleasetoolkit/versioning/formatters/derived_build_code_formatter.rb

```diff
@@ -19,15 +28,55 @@ class DerivedBuildCodeFormatter
         # @return [String] The formatted build code string.
         #
         def build_code(build_code = nil, version:)
-          format(
-            # 1 is appended to the beginning of the string in case there needs to be additional platforms or
-            # extensions that could then use a different digit prefix such as 2, etc.
-            '1%<major>.2i%<minor>.2i%<patch>.2i%<build_number>.2i',
+          result = format(
+            # The prefix is configurable to allow for additional platforms or
+            # extensions that could use a different digit prefix such as 2, etc.
+            '%<prefix>s%<major>.2i%<minor>.2i%<patch>.2i%<build_number>.2i',
+            prefix: @prefix,
             major: version.major,
             minor: version.minor,
             patch: version.patch,
             build_number: version.build_number
           )
+
+          # Only trim leading zeros when prefix is empty (to avoid removing explicit "0" prefix)
+          if @prefix.empty?
+            result.gsub(/^0+/, '')
+          else
+            result
+          end
```

Comment:
I'm actually not sure how the Google Play API would react to submitting a build with a `versionCode` that has a leading zero 🤔